src/ja/server/dispatcher/Dispatcher.py

```diff
@@ -0,0 +1,43 @@
+from ja.common.job import Job
+from ja.server.proxy.proxy import WorkerProxy
+from typing import List
+from ja.server.database.database import ServerDatabase
+
+
+JobDistribution = ServerDatabase.JobDistribution
+
+
+class Dispatcher:
+    """
+    Represents the mediator between the Scheduler and the work machines.
+    """
+
+    def __init__(self, worker_proxies: List[WorkerProxy]):
+        """!
+        Constructor for the dispatcher class.
+        @param worker_proxies proxies of the work machines.
+        """
+
+    def set_distribution(self, job_distribution: JobDistribution) -> None:
+        """!
+        Distributes the newly added jobs to the work machines.
+        and potentially pauses jobs that need to be paused.
+
+        @param jobs list of the new job distribution.
+        """
+
+    def _get_changed_jobs(self, jobs: List[Job]) -> List[Job]:
+        """!
+        @return list of the jobs whose state has been changed.
```

Comment:
Changed relative to what?
The last time the set_distribution was called?

Reply:
yes